packages/venia-concept/src/components/MiniCart/product.js

```diff
@@ -47,23 +72,33 @@ class Product extends Component {
     }
 
     get options() {
-        const { classes, item } = this.props;
-
-        return item.options && item.options.length > 0 ? (
-            <dl className={this.props.classes.options}>
-                {item.options.map(({ name, value }) => (
-                    <Fragment key={name}>
-                        <dt className={classes.optionName}>{name}</dt>
-                        <dd className={classes.optionValue}>{value}</dd>
-                    </Fragment>
-                ))}
-            </dl>
-        ) : null;
+        const { classes, item, totalsItems } = this.props;
+        const { item_id } = item;
+        const regex = /[\[\]']+/g; // regex to to remove [] , use it to check if options are really empty
+        return totalsItems
+            .filter(totalsItem => totalsItem.item_id === item_id)
+            .map(totalsItem => {
+                const { options } = totalsItem;
+                return options && options.replace(regex, '').length > 0 ? (
```

Comment:
I'm a little confused by the need for `regex`. Is `options` a string or an array?

Looking at [the Magento REST API documentation](https://devdocs.magento.com/swagger/) (`GET /V1/guest-carts/{cartId}/totals`), it should be a `string`:

```
"options": "string",
```

This code makes me assume that `options` can sometimes be a stringified version of an array? With values like "['foo', 'bar', 'baz']" ?

If that's the case, and you're trying to determine if options is empty as the comment on line 77 indicates, I think a simple `options === '[]'` could reduce the complexity here.

In other words, this line could become

```js
return options && options !== '[]'
```

and we could drop the regex.

Alternately, you could `JSON.parse(options)` ahead of time and use the parsed version throughout this section. That might be even better.

Reply:
yeah much better, updated accordingly, thx for the suggestions!